.github/workflows/distribute-grain.yml

```diff
@@ -0,0 +1,59 @@
+name: Distribute Grain
+on:
+  # A new Cred interval every Sunday. We'll distribute Grain 5 minutes
+  # after the new interval
+  schedule:
+    - cron: 5 0 * * 0 # 00:05 UTC, 16:05 PST
+  push:
+    branches:
+      - "graintest-*" # won't actually trigger GitHub Pages deploy
+
+jobs:
+  distribute-grain:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false # Required to make github pages deployment work correctly
+
+      - name: "[debug] Check GitHub event name"
+        # This is used to determine whether we actually deploy to GitHub
+        # Pages, so we print it upfront to aid with debugging.
+        run: echo ${{ github.event_name }}
+
+      - name: Install Packages 🔧
+        run: yarn --frozen-lockfile
+
+      - name: Load Data into Graph 🔌
+        run: |
+          yarn load
+          yarn graph
+        env:
+          SOURCECRED_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SOURCECRED_DISCORD_TOKEN: ${{ secrets.SOURCECRED_DISCORD_TOKEN }}
+
+      - name: Compute Cred 🧮
+        run: yarn score
+
+      - name: Distribute Grain 💸
+        run: yarn grain
+
+      - name: Commit Ledger 💾
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Auto-update ledger
+          file_pattern: data/ledger.json
```

Comment:
As configured right now, this push will fail because no credentials are configured in git: 

https://github.com/sourcecred/cred/blob/cc80663841ec10d21d43cc11aa6370ffb701b21a/.github/workflows/distribute-grain.yml#L17

I think we might want to try persisting credentials at checkout, writing a manual push, then clearing the credentials, either via a second checkout step or something more explicit

Reply:
Got it, thanks (I see what you were talking about now). I didn’t realize
that this was also meant to push, since that’s not in the name or
description—a commit is not a push. Agreed that pushing and committing
explicitly would be clearer. I’ll leave that up to you, since you’ve
been working on it. I’ve updated the workflow to skip this action on
non-schedule runs, just like the GitHub Pages deploy.